x-pack/plugins/security_solution/common/detection_engine/rule_monitoring/model/execution_event.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import * as t from 'io-ts';
+import { enumeration, IsoDateString } from '@kbn/securitysolution-io-ts-types';
+import { enumFromString } from '../../../utils/enum_from_string';
+import { TLogLevel } from './log_level';
+
+/**
+ * Type of a plain rule execution event.
+ */
+export enum RuleExecutionEventType {
+  /**
+   * Simple log message of some log level, such as debug, info or error.
+   */
+  'message' = 'message',
+
+  /**
+   * We log an event of this type each time a rule changes its status during an execution.
+   */
+  'status-change' = 'status-change',
+
+  /**
+   * We log an event of this type at the end of a rule execution. It contains various execution
+   * metrics such as search and indexing durations.
+   */
+  'execution-metrics' = 'execution-metrics',
```

Comment:
For some reason, the execution events table doesn't show any execution metrics events:
![Screenshot 2022-07-22 at 10 50 49](https://user-images.githubusercontent.com/1938181/180405541-7118dd20-c20f-4555-ac7f-3a3304fd203b.png)

While there are some metrics displayed on the execution result table. At least there is the search duration metric visible:
![Screenshot 2022-07-22 at 10 48 09](https://user-images.githubusercontent.com/1938181/180405506-09b1468b-dc79-4ff0-b6d1-90508b567a0e.png)

Reply:
Yeah, this looks like a bug. I'll try to fix it 👍

Reply:
Apparently, the reason is in the inclusion of `event.kind:event` filter when fetching events:

```ts
      const kqlFilter = kqlAnd([
        `event.provider:${RULE_EXECUTION_LOG_PROVIDER}`,
        'event.kind:event',
        eventTypes.length > 0 ? `event.action:(${kqlOr(eventTypes)})` : '',
        logLevels.length > 0 ? `log.level:(${kqlOr(logLevels)})` : '',
      ]);
```

I still have to verify this.